test/Microsoft.ML.Tests/TrainerEstimators/SdcaTests.cs

```diff
@@ -88,11 +88,66 @@ public void SdcaLogisticRegression()
             Assert.InRange(first.Probability, 0.8, 1);
         }
 
+        [Fact]
+        public void SdcaLogisticRegressionWithWeight()
+        {
+            // Generate C# objects as training examples.
+            var rawData = SamplesUtils.DatasetUtils.GenerateBinaryLabelFloatFeatureVectorFloatWeightSamples(100);
+
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            var mlContext = new MLContext(0);
+
+            // Read the data as an IDataView.
+            var data = mlContext.Data.LoadFromEnumerable(rawData);
+
+            // ML.NET doesn't cache data set by default. Caching is very helpful when working with iterative
+            // algorithms which needs many data passes. Since SDCA is the case, we cache.
+            data = mlContext.Data.Cache(data);
+
+            // Verify SdcaLogisticRegression with and without weights.
+            var sdcaWithoutWeightBinary = mlContext.BinaryClassification.Trainers.SdcaLogisticRegression(
+                new SdcaLogisticRegressionBinaryTrainer.Options { NumberOfThreads = 1 });
+            var sdcaWithWeightBinary = mlContext.BinaryClassification.Trainers.SdcaLogisticRegression(
+                new SdcaLogisticRegressionBinaryTrainer.Options { ExampleWeightColumnName = "Weight", NumberOfThreads = 1 });
+
+            var prediction1 = sdcaWithoutWeightBinary.Fit(data).Transform(data);
```

Comment:
Could you check that if all (or most) results in `prediction1` and `prediction2` are different?

Reply:
added checks to ensure the model parameters for both models are different

---
In reply to: [270206386](https://github.com/dotnet/machinelearning/pull/3104#discussion_r270206386) [](ancestors = 270206386)